pkg/cloudprovider/providers/azure/azure_zones.go

```diff
@@ -61,14 +62,31 @@ func (az *Cloud) GetZone() (cloudprovider.Zone, error) {
 // This is particularly useful in external cloud providers where the kubelet
 // does not initialize node data.
 func (az *Cloud) GetZoneByProviderID(providerID string) (cloudprovider.Zone, error) {
-	return cloudprovider.Zone{}, errors.New("GetZoneByProviderID not implemented")
+	nodeName, err := splitProviderID(providerID)
```

Comment:
Not a huge fan of this function name `splitProviderID`, but seems outside the scope of this PR :P

Reply:
yeah, not for now